addons/website_livechat/controllers/webclient.py

```diff
@@ -0,0 +1,59 @@
+from odoo import Command
+from odoo.addons.mail.controllers.webclient import WebclientController
+from odoo.addons.mail.tools.discuss import Store
+
+
+class WebClient(WebclientController):
+    def _process_request_for_all(self, store: Store, name, params):
+        if name == "init_livechat" and (
+            chat_request_channel := self._link_visitor_to_channel(params)
+        ):
+            store.add(chat_request_channel, extra_fields={"open_chat_window": True})
+            chat_request_channel.is_pending_chat_request = False
+        super()._process_request_for_all(store, name, params)
+
+    def _link_visitor_to_channel(self, channel_id):
```

Comment:
It's not clear from name that we are talking about a livechat_channel and not a discuss_channel. We should always clarify I think.
And maybe we should rename the livechat one to something else eventually.